src/Controller/SignupController.php

```diff
@@ -25,31 +27,31 @@ public function __construct(ViewRenderer $viewRenderer)
     }
 
     public function signup(
-        RequestInterface $request,
-        IdentityRepositoryInterface $identityRepository,
         EntityWriter $entityWriter,
+        IdentityRepositoryInterface $identityRepository,
+        RequestInterface $request,
+        ResponseFactoryInterface $responseFactory,
+        TranslatorInterface $translator,
         UrlGeneratorInterface $urlGenerator,
-        LoggerInterface $logger,
-        ResponseFactoryInterface $responseFactory
+        ValidatorInterface $validator
     ): ResponseInterface {
         $body = $request->getParsedBody();
         $error = null;
 
-        if ($request->getMethod() === Method::POST) {
-            try {
-                foreach (['login', 'password'] as $name) {
-                    if (empty($body[$name])) {
-                        throw new \InvalidArgumentException(ucfirst($name) . ' is required.');
-                    }
-                }
+        $loginForm = new LoginForm($translator);
```

Comment:
I don't think we should reuse login form for both signup and login. Should be two separate forms.

Usually a signup form has more data: password repeat, some user data etc.